app/api/login+api.ts

```diff
@@ -0,0 +1,104 @@
+import crypto from 'node:crypto';
+import bcryptJs from 'bcryptjs';
+import { createSessionInsecure } from '../../database/sessions';
+import { getUserWithPasswordHashInsecure } from '../../database/users';
+import { ExpoApiResponse } from '../../ExpoApiResponse';
+import { type User, userSchema } from '../../migrations/00001-createTableUsers';
+import { createSerializedRegisterSessionTokenCookie } from '../../util/cookies';
+
+export type LoginResponseBodyPost =
+  | {
+      user: { username: User['username'] };
+    }
+  | { error: string; errorIssues?: { message: string }[] };
+
+export async function POST(
+  request: Request,
+): Promise<ExpoApiResponse<LoginResponseBodyPost>> {
+  // Task: Implement the user login workflow with session
+
+  // 1. Get the user data from the request
+  const requestBody = await request.json();
+
+  // 2. Validate the user data with zod
+  const result = userSchema.safeParse(requestBody);
+
+  if (!result.success) {
+    return ExpoApiResponse.json(
+      {
+        error: 'Request does not contain user object',
+        errorIssues: result.error.issues,
+      },
+      {
+        status: 400,
+      },
+    );
+  }
+
+  // 3. verify the user credentials
+  const userWithPasswordHash = await getUserWithPasswordHashInsecure(
+    result.data.username,
+  );
+
+  if (!userWithPasswordHash) {
+    return ExpoApiResponse.json(
+      {
+        error: 'username or password not valid',
```

Comment:
Capitalization

(for all error messages in the PR)

Reply:
d7d9acacd628969d1ab034d3b7cb87d34f10d1c6